### ISSUE TITLE
Failing to find Subscription ID when targeting AzureUSGovernment Tenant

### ISSUE THREAD

**Issue by adrian-gonzalez** [Azure/azure-sdk-for-python#27038]
- **Package Name**:  MLClient
- **Package Version**:  SDK V2
- **Operating System**:  AML Compute Instance STANDARD_DS11_V2
- **Python Version**: Python 3.10

**Describe the bug**
After initializing an instance of the MLClient module, executing any of it's methods results in the error below.

**To Reproduce**

Pre-requirements:
1. Have an AzureUSGovernment tenant and subscrition
2. Have an AML Workspace created, along with a Compute Instance
3. Have a Service Principal created in the above subscription, and given a "Contributor" role assignment to the AML Workspace
4. Run a notebook in AML using the compute instance, and updating the placeholder environment variables:

```
from azure.ai.ml.entities import AmlCompute
from azure.ai.ml import MLClient
from azure.identity import DefaultAzureCredential, AzureAuthorityHosts, EnvironmentCredential

import traceback

# Set ENV Variables

os.environ["AZURE_CLIENT_SECRET"] = "<value>"
os.environ["AZURE_CLIENT_ID"] = "<value>"
os.environ["AZURE_TENANT_ID"] = "<value>"
os.environ["AZURE_AUTHORITY_HOST"] = AzureAuthorityHosts.AZURE_GOVERNMENT


credentials = DefaultAzureCredential(
    interactive_browser_tenant_id=os.environ["AZURE_TENANT_ID"],
    authority=AzureAuthorityHosts.AZURE_GOVERNMENT
    )

ml_client = MLClient(
    credential=credentials,
    subscription_id="<value>",
    resource_group_name="<value>",
    workspace_name="<value>",
    cloud="AzureUSGovernment",
)

# Name assigned to the compute cluster
cpu_compute_target = "cpu-cluster-2"

try:
    # let's see if the compute target already exists
    cpu_cluster = ml_client.compute.get(cpu_compute_target)
    print(
        f"You already have a cluster named {cpu_compute_target}, we'll reuse it as is."
    )

except Exception:
    print("Creating a new cpu compute target...")

    # Let's create the Azure ML compute object with the intended parameters
    cpu_cluster = AmlCompute(
        name=cpu_compute_target,
        # Azure ML Compute is the on-demand VM service
        type="amlcompute",
        # VM Family
        size="STANDARD_DS3_V2",
        # Minimum running nodes when there is no job running
        min_instances=0,
        # Nodes in cluster
        max_instances=4,
        # How many seconds will the node running after the job termination
        idle_time_before_scale_down=180,
        # Dedicated or LowPriority. The latter is cheaper but there is a chance of job termination
        tier="Dedicated",
    )

    # Now, we pass the object to MLClient's create_or_update method
    cpu_cluster = ml_client.compute.begin_create_or_update(cpu_cluster)

print(
    f"AMLCompute with name {cpu_cluster.name} is created, the compute size is {cpu_cluster.size}"
)
```

**Expected behavior**
The above code should result in either a new CPU Cluster being created, or printing out the message `You already have a cluster named {cpu_compute_target}, we'll reuse it as is."`

**Screenshots**

The actual behavior is an error:
```
ResourceNotFoundError: (SubscriptionNotFound) The subscription 'xxxxxxxxxxxxxxxx' could not be found.
Code: SubscriptionNotFound
Message: The subscription 'xxxxxxxxxxxxxxxx' could not be found.
```

The stack trace is:

```
Creating a new cpu compute target...
---------------------------------------------------------------------------
ResourceNotFoundError                     Traceback (most recent call last)
Input In [8], in <cell line: 13>()
     13 try:
     14     # let's see if the compute target already exists
---> 15     cpu_cluster = ml_client_6.compute.get(cpu_compute_target)
     16     print(
     17         f"You already have a cluster named {cpu_compute_target}, we'll reuse it as is."
     18     )

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_telemetry/activity.py:169, in monitor_with_activity.<locals>.monitor.<locals>.wrapper(*args, **kwargs)
    168 with log_activity(logger, activity_name or f.__name__, activity_type, custom_dimensions):
--> 169     return f(*args, **kwargs)

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_operations/compute_operations.py:75, in ComputeOperations.get(self, name)
     67 """Get a compute resource
     68 
     69 :param name: Name of the compute
   (...)
     72 :rtype: Compute
     73 """
---> 75 response, rest_obj = self._operation.get(
     76     self._operation_scope.resource_group_name,
     77     self._workspace_name,
     78     name,
     79     cls=get_http_response_and_deserialized_from_pipeline_response,
     80 )
     81 # TODO: Remove warning logging after 05/31/2022 (Task 1776012)

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/core/tracing/decorator.py:83, in distributed_trace.<locals>.decorator.<locals>.wrapper_use_tracer(*args, **kwargs)
     82 if span_impl_type is None:
---> 83     return func(*args, **kwargs)
     85 # Merge span is parameter is set, but only if no explicit parent are passed

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_restclient/v2022_01_01_preview/operations/_compute_operations.py:577, in ComputeOperations.get(self, resource_group_name, workspace_name, compute_name, **kwargs)
    576 if response.status_code not in [200]:
--> 577     map_error(status_code=response.status_code, response=response, error_map=error_map)
    578     error = self._deserialize.failsafe_deserialize(_models.ErrorResponse, pipeline_response)

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/core/exceptions.py:105, in map_error(status_code, response, error_map)
    104 error = error_type(response=response)
--> 105 raise error

ResourceNotFoundError: (SubscriptionNotFound) The subscription '50ff9458-6372-4522-8227-327043deaef5' could not be found.
Code: SubscriptionNotFound
Message: The subscription '50ff9458-6372-4522-8227-327043deaef5' could not be found.

During handling of the above exception, another exception occurred:

ResourceNotFoundError                     Traceback (most recent call last)
Input In [8], in <cell line: 13>()
     24     cpu_cluster = AmlCompute(
     25         name=cpu_compute_target,
     26         # Azure ML Compute is the on-demand VM service
   (...)
     37         tier="Dedicated",
     38     )
     40     # Now, we pass the object to MLClient's create_or_update method
---> 41     cpu_cluster = ml_client_6.compute.begin_create_or_update(cpu_cluster)
     43 print(
     44     f"AMLCompute with name {cpu_cluster.name} is created, the compute size is {cpu_cluster.size}"
     45 )

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_telemetry/activity.py:169, in monitor_with_activity.<locals>.monitor.<locals>.wrapper(*args, **kwargs)
    166 @functools.wraps(f)
    167 def wrapper(*args, **kwargs):
    168     with log_activity(logger, activity_name or f.__name__, activity_type, custom_dimensions):
--> 169         return f(*args, **kwargs)

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_operations/compute_operations.py:116, in ComputeOperations.begin_create_or_update(self, compute, **kwargs)
    107 @monitor_with_activity(logger, "Compute.BeginCreateOrUpdate", ActivityType.PUBLICAPI)
    108 def begin_create_or_update(self, compute: Compute, **kwargs: Any) -> LROPoller:
    109     """Create a compute
    110 
    111     :param compute: Compute definition.
   (...)
    114     :rtype: LROPoller
    115     """
--> 116     compute.location = self._get_workspace_location()
    117     compute._set_full_subnet_name(self._operation_scope.subscription_id, self._operation_scope.resource_group_name)
    119     compute_rest_obj = compute._to_rest_object()

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_operations/compute_operations.py:308, in ComputeOperations._get_workspace_location(self)
    307 def _get_workspace_location(self) -> str:
--> 308     workspace = self._workspace_operations.get(self._resource_group_name, self._workspace_name)
    309     return workspace.location

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/core/tracing/decorator.py:83, in distributed_trace.<locals>.decorator.<locals>.wrapper_use_tracer(*args, **kwargs)
     81 span_impl_type = settings.tracing_implementation()
     82 if span_impl_type is None:
---> 83     return func(*args, **kwargs)
     85 # Merge span is parameter is set, but only if no explicit parent are passed
     86 if merge_span and not passed_in_parent:

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_restclient/v2022_01_01_preview/operations/_workspaces_operations.py:615, in WorkspacesOperations.get(self, resource_group_name, workspace_name, **kwargs)
    612 response = pipeline_response.http_response
    614 if response.status_code not in [200]:
--> 615     map_error(status_code=response.status_code, response=response, error_map=error_map)
    616     error = self._deserialize.failsafe_deserialize(_models.ErrorResponse, pipeline_response)
    617     raise HttpResponseError(response=response, model=error, error_format=ARMErrorFormat)

File /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/core/exceptions.py:105, in map_error(status_code, response, error_map)
    103     return
    104 error = error_type(response=response)
--> 105 raise error

```

**Additional context**

I looked through the source code in [_azure_environments.py](https://github.com/Azure/azure-sdk-for-python/blob/azure-ai-ml_1.0.0/sdk/ml/azure-ai-ml/azure/ai/ml/_azure_environments.py) file and also the [_ml_client.py](https://github.com/Azure/azure-sdk-for-python/blob/azure-ai-ml_1.0.0/sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py) file to infer what environment variables and values I needed to pass into the MLClient constructor.  However, something doesn't appear to be working correctly.

**Comment by harneetvirk** [comment]
Here is an example for running a notebook in Non-public cloud: https://github.com/Azure/azureml-examples/blob/main/sdk/python/jobs/multicloud-configuration.ipynb

You may need to pass the cloud name in the kwargs for MLClient.

    # NOTE: cloud parameter is required in kwargs to signal mlclient to connect to the appropriate endpoints in Azure.
    kwargs = {"cloud": "AzureChinaCloud"}
    ml_client = MLClient(credential, subscription_id, resource_group, **kwargs)

**Comment by adrian-gonzalez** [comment]
Hi @harneetvirk , unfortunately I am still facing the same error.

Below is my latest attempt.

First I make sure that the authentication works (replacing the placeholder values with the correct values)

```
from azure.ai.ml.entities import AmlCompute
from azure.ai.ml import MLClient
from azure.identity import DefaultAzureCredential, AzureAuthorityHosts

# Set ENV Variables

os.environ["AZURE_USERNAME"] = "xxxxxxxxx"
os.environ["AZURE_PASSWORD"] = "xxxxxxxxx"
os.environ["AZURE_TENANT_ID"] = "xxxxxxxxxx"

kwargs = {"cloud": "AzureUSGovernment"}
credentials = DefaultAzureCredential(authority=AzureAuthorityHosts.AZURE_GOVERNMENT)

credentials.get_token("https://management.usgovcloudapi.net/.default")

```
This results in a successful authentication and token retrieval. 

But then running the below command, after putting in the correct subscription ID, using the ML Client errors out.

```

ml_client = MLClient(
    credential=credentials,
    subscription_id="xxxxxxxxx",
    resource_group_name="xxxxxxxx",
    workspace_name="xxxxxxx",
    **kwargs,
)

print(ml_client)


# Get a list of workspaces in a resource group
for ws in ml_client.workspaces.list():
    print(ws.name, ":", ws.location, ":", ws.description)
```

error is 
```
ResourceNotFoundError: (SubscriptionNotFound) The subscription '50ff9458-6372-4522-8227-327043deaef5' could not be found.
Code: SubscriptionNotFound
Message: The subscription '50ff9458-6372-4522-8227-327043deaef5' could not be found.
```

**Comment by kottofy** [comment]
I can perform the following in an Azure US Gov ML Studio notebook targeting Python 3.10 - SDK V2 (default Compute) and it works.

Seems like there is a default version installed of `azure-ai-ml` at 2.4.1 that might be causing some issues.

---------CELL 1---------

`pip list | grep azure`


Output:
azure-ai-ml              2.4.1
azure-common             1.1.28
azure-core               1.22.1
azure-identity           1.10.0
azure-mgmt-core          1.3.0
azure-ml                 2.3.1
azure-storage-blob       12.9.0
azure-storage-file-share 12.7.0
Note: you may need to restart the kernel to use updated packages.

---------CELL 2---------

`pip uninstall azure-ai-ml azure-ml -y`


Output:

Found existing installation: azure-ai-ml 2.4.1
Uninstalling azure-ai-ml-2.4.1:
  Successfully uninstalled azure-ai-ml-2.4.1
Found existing installation: azure-ml 2.3.1
Uninstalling azure-ml-2.3.1:
  Successfully uninstalled azure-ml-2.3.1
Note: you may need to restart the kernel to use updated packages.

---------CELL 3---------

`pip install --pre azure-ai-ml`


Output:

Collecting azure-ai-ml
  Downloading azure_ai_ml-1.0.0-py3-none-any.whl (4.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.0/4.0 MB 55.9 MB/s eta 0:00:00:00:010:01
Collecting azure-storage-blob<13.0.0,>=12.10.0
  Downloading azure_storage_blob-12.14.1-py3-none-any.whl (383 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 383.2/383.2 kB 28.9 MB/s eta 0:00:00
Requirement already satisfied: azure-mgmt-core<2.0.0,>=1.3.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (1.3.0)
Requirement already satisfied: azure-common<2.0.0,>=1.1 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (1.1.28)
Requirement already satisfied: msrest>=0.6.18 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (0.6.21)
Collecting strictyaml<=1.6.1
  Downloading strictyaml-1.6.1.tar.gz (137 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 137.7/137.7 kB 11.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Requirement already satisfied: azure-core!=1.22.0,<2.0.0,>=1.8.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (1.22.1)
Requirement already satisfied: pydash<6.0.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (4.9.0)
Requirement already satisfied: pyyaml<7.0.0,>=5.1.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (6.0)
Requirement already satisfied: colorama<=0.4.4 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (0.4.4)
Collecting azure-storage-file-datalake<13.0.0
  Downloading azure_storage_file_datalake-12.9.1-py3-none-any.whl (238 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 238.8/238.8 kB 23.0 MB/s eta 0:00:00
Requirement already satisfied: tqdm<=4.63.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (4.63.0)
Requirement already satisfied: pyjwt<3.0.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (2.3.0)
Requirement already satisfied: marshmallow<4.0.0,>=3.5 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (3.17.0)
Requirement already satisfied: typing-extensions<5.0.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (3.10.0.2)
Requirement already satisfied: azure-storage-file-share<13.0.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (12.7.0)
Requirement already satisfied: isodate in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (0.6.1)
Requirement already satisfied: jsonschema<5.0.0,>=4.0.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-ai-ml) (4.13.0)
Requirement already satisfied: six>=1.11.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-core!=1.22.0,<2.0.0,>=1.8.0->azure-ai-ml) (1.16.0)
Requirement already satisfied: requests>=2.18.4 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-core!=1.22.0,<2.0.0,>=1.8.0->azure-ai-ml) (2.28.1)
Requirement already satisfied: cryptography>=2.1.4 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from azure-storage-blob<13.0.0,>=12.10.0->azure-ai-ml) (37.0.4)
Collecting msrest>=0.6.18
  Downloading msrest-0.7.1-py3-none-any.whl (85 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.4/85.4 kB 10.9 MB/s eta 0:00:00
Collecting azure-core!=1.22.0,<2.0.0,>=1.8.0
  Downloading azure_core-1.26.0-py3-none-any.whl (178 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 178.9/178.9 kB 21.4 MB/s eta 0:00:00
Collecting typing-extensions<5.0.0
  Downloading typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from jsonschema<5.0.0,>=4.0.0->azure-ai-ml) (0.18.1)
Requirement already satisfied: attrs>=17.4.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from jsonschema<5.0.0,>=4.0.0->azure-ai-ml) (22.1.0)
Requirement already satisfied: packaging>=17.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from marshmallow<4.0.0,>=3.5->azure-ai-ml) (21.3)
Requirement already satisfied: requests-oauthlib>=0.5.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from msrest>=0.6.18->azure-ai-ml) (1.3.1)
Requirement already satisfied: certifi>=2017.4.17 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from msrest>=0.6.18->azure-ai-ml) (2022.6.15)
Requirement already satisfied: python-dateutil>=2.6.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from strictyaml<=1.6.1->azure-ai-ml) (2.8.2)
Requirement already satisfied: cffi>=1.12 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from cryptography>=2.1.4->azure-storage-blob<13.0.0,>=12.10.0->azure-ai-ml) (1.15.1)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from packaging>=17.0->marshmallow<4.0.0,>=3.5->azure-ai-ml) (3.0.9)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from requests>=2.18.4->azure-core!=1.22.0,<2.0.0,>=1.8.0->azure-ai-ml) (1.26.11)
Requirement already satisfied: charset-normalizer<3,>=2 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from requests>=2.18.4->azure-core!=1.22.0,<2.0.0,>=1.8.0->azure-ai-ml) (2.1.0)
Requirement already satisfied: idna<4,>=2.5 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from requests>=2.18.4->azure-core!=1.22.0,<2.0.0,>=1.8.0->azure-ai-ml) (3.3)
Requirement already satisfied: oauthlib>=3.0.0 in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from requests-oauthlib>=0.5.0->msrest>=0.6.18->azure-ai-ml) (3.2.0)
Requirement already satisfied: pycparser in /anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages (from cffi>=1.12->cryptography>=2.1.4->azure-storage-blob<13.0.0,>=12.10.0->azure-ai-ml) (2.21)
Building wheels for collected packages: strictyaml
  Building wheel for strictyaml (setup.py) ... done
  Created wheel for strictyaml: filename=strictyaml-1.6.1-py3-none-any.whl size=123931 sha256=7f10357971c55b3c29d2dbee29b816db58830ad42ac89258d28bd87636c1f5a7
  Stored in directory: /home/azureuser/.cache/pip/wheels/fb/ca/49/3c5046dee736c4c938048ce89b236b1643ea83178517b5f88a
Successfully built strictyaml
Installing collected packages: typing-extensions, strictyaml, azure-core, msrest, azure-storage-blob, azure-storage-file-datalake, azure-ai-ml
  Attempting uninstall: typing-extensions
    Found existing installation: typing-extensions 3.10.0.2
    Uninstalling typing-extensions-3.10.0.2:
      Successfully uninstalled typing-extensions-3.10.0.2
  Attempting uninstall: azure-core
    Found existing installation: azure-core 1.22.1
    Uninstalling azure-core-1.22.1:
      Successfully uninstalled azure-core-1.22.1
  Attempting uninstall: msrest
    Found existing installation: msrest 0.6.21
    Uninstalling msrest-0.6.21:
      Successfully uninstalled msrest-0.6.21
  Attempting uninstall: azure-storage-blob
    Found existing installation: azure-storage-blob 12.9.0
    Uninstalling azure-storage-blob-12.9.0:
      Successfully uninstalled azure-storage-blob-12.9.0
Successfully installed azure-ai-ml-1.0.0 azure-core-1.26.0 azure-storage-blob-12.14.1 azure-storage-file-datalake-12.9.1 msrest-0.7.1 strictyaml-1.6.1 typing-extensions-4.4.0
Note: you may need to restart the kernel to use updated packages.

---------CELL 4---------

```python
import logging
import requests
import os
from azure.ai.ml import MLClient
from azure.identity import AzureAuthorityHosts, DefaultAzureCredential
from azure.ai.ml.entities import Workspace

subscription_id = "YOUR_VALUE_HERE"
resource_group = "YOUR_VALUE_HERE"
workspace_name = "YOUR_VALUE_HERE"

logging.basicConfig()
logging.getLogger().setLevel(logging.DEBUG)

try:
    credential = DefaultAzureCredential(authority=AzureAuthorityHosts.AZURE_GOVERNMENT)
except Exception as ex:
    raise ex
print(credential)

try:
    kwargs = {"cloud": "AzureUSGovernment"}
    ml_client = MLClient(credential, subscription_id, resource_group, **kwargs)
except Exception as ex:
    raise ex

print(ml_client)


# Get a list of workspaces in a resource group
for ws in ml_client.workspaces.list():
    print(ws.name, ":", ws.location, ":", ws.description)
```

**Comment by harneetvirk** [comment]
Could you please try to login to azure cli from the same machine from where you are running the notebooks and set the default subscription?
```
az cloud set -n AzureUSGovernment
az account set -s <SUBSCRIPTION-ID>

````

This should set the default subscription for you on the machine.

I have tried the following code snippet in Government cloud, and this is working with azure-ai-ml==1.0.0.
Had to update the call to create compute by appending .result() for LRO poller.

```
from azure.ai.ml.entities import AmlCompute
from azure.ai.ml import MLClient
from azure.identity import DefaultAzureCredential, AzureAuthorityHosts

import traceback

# Enter details of your subscription
subscription_id = "SOME-SUBSCRITION-ID-IN-GOVT-CLOUD"
resource_group = "test-rg-221005"
workspace_name = "est-usgovvirginia"
credentials = DefaultAzureCredential(authority=AzureAuthorityHosts.AZURE_GOVERNMENT)

kwargs = {"cloud": "AzureUSGovernment"}
ml_client = MLClient(credential, subscription_id, resource_group, **kwargs)
    
ml_client = MLClient(
    credential=credentials,
    subscription_id=subscription_id,
    resource_group_name=resource_group,
    workspace_name=workspace_name,
    cloud="AzureUSGovernment",
)
# Name assigned to the compute cluster
cpu_compute_target = "cpu-cluster-3"

try:
    # let's see if the compute target already exists
    cpu_cluster = ml_client.compute.get(cpu_compute_target)
    print(
        f"You already have a cluster named {cpu_compute_target}, we'll reuse it as is."
    )

except Exception:
    print("Creating a new cpu compute target...")

    # Let's create the Azure ML compute object with the intended parameters
    cpu_cluster = AmlCompute(
        name=cpu_compute_target,
        # Azure ML Compute is the on-demand VM service
        type="amlcompute",
        # VM Family
        size="STANDARD_DS3_V2",
        # Minimum running nodes when there is no job running
        min_instances=0,
        # Nodes in cluster
        max_instances=4,
        # How many seconds will the node running after the job termination
        idle_time_before_scale_down=180,
        # Dedicated or LowPriority. The latter is cheaper but there is a chance of job termination
        tier="Dedicated",
    )

    # Now, we pass the object to MLClient's create_or_update method
    cpu_cluster = ml_client.compute.begin_create_or_update(cpu_cluster).result()

print(
    f"AMLCompute with name {cpu_cluster.name} is created, the compute size is {cpu_cluster.size}"
)
```
Returned the following:
Creating a new cpu compute target...
AMLCompute with name cpu-cluster-3 is created, the compute size is STANDARD_DS3_V2

**Comment by harneetvirk** [comment]
@adrian-gonzalez : I created a new Conda environment and installed the GA version of SDK (1.0.0) and tried running the notebook in usgovvirginia region to repro this issue but unfortunately, I was able to able to reproduce this issue after I configure the cloud name and account using CLI.

```
az cloud set -n AzureUSGovernment
az account set -s <SUBSCRIPTION-ID>
```
By default, the SDK or CLI tries to connect to public cloud.

**Comment by adrian-gonzalez** [comment]
Hi @harneetvirk  -

Please read through our previous comments. 

The steps to reproduce the issue lies with azure-ai-ml v2.4.1.

it is this version that is the default when we are creating an instance of Azure Machine Learning, and therefore preventing our team from using this python package.

**Comment by harneetvirk** [comment]
azure-ai-ml v2.4.1 is having some known issues and will be removed from CI from the next release and will be replaced with azure-ai-ml 1.0.0 from pypi. To unblock, please install azure-ai-ml 1.0.0 from pypi.

**Comment by adrian-gonzalez** [comment]
Thank you @harneetvirk . We can do that as a temporary workaround. 

When is the next release slated for?

Can you confirm if once azure-ai-ml v2.4.1 is removed from CI, whether newly created AML instance won't have this issue moving forward? 

Just want to be sure that from a developer experience, that teams creating new AML instances don't have to always manually downgrade the auzre-ai-ml package to 1.0.0

**Comment by adrian-gonzalez** [comment]
@xiangyan99  can you or @harneetvirk provide guidance on the remaining above questions? I would like to confirm that the issue is resolved and that the steps to reproduce no longer result in the issue prior to closing this issue out.

**Comment by xiangyan99** [comment]
@adrian-gonzalez if you don't mind, could you open a new issue with the questions?

Thanks.

**Comment by adrian-gonzalez** [comment]
@xiangyan99  That doesn't seem efficient, are you sure we want to go that route? It'll effectively a copy of this issue to not duplicate the steps to reproduce and discussions.

Happy to do that if that's the approach

**Comment by adrian-gonzalez** [comment]
@xiangyan99 can you or @harneetvirk provide guidance on the remaining above questions? I would like to confirm that the issue is resolved and that the steps to reproduce no longer result in the issue prior to closing this issue out.

**Comment by harneetvirk** [comment]
The new CI image has been released with SDK v2 package installed from pypi. Please create a new Compute Instance.

**Comment by adrian-gonzalez** [comment]
Thanks @harneetvirk!
